### ISSUE TITLE
Supported html placeholder in a transport rule

### ISSUE THREAD

**Issue by riohedfield** [MicrosoftDocs/office-docs-powershell#10710]
Hello, 

I came cross this article specifically the following parameters:
-GenerateNotification 
-ApplyHtmlDisclaimerText

For -GenerateNotification it's clearly mentioned:
This parameter supports plain text, HTML tags and the following keywords that use values from the original message:

%%From%%
%%To%%
%%Cc%%
%%Subject%%
%%Headers%%
%%MessageDate%%

Now for -ApplyHtmlDisclaimerText when I try to add a html code with a placeholder %%From%% or %%FromAddress%% or %%Email%% to generate a disclaimer that is supposed to pull out the sender's address from the original message and add it to a text ( then Prepend the disclaimer) the GUI is returning this error:
System.ArgumentException|The disclaimer text contains an invalid macro name: 'From'.

This is the full error when I try to save the transport rule:
![image](https://user-images.githubusercontent.com/59142439/230867083-3b921d11-3776-4596-99eb-209e509fa6b8.png)

Can you please update the documentation with what is exactly supported for -ApplyHtmlDisclaimerText ?

BR
Ribo

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 11bf4f19-ce5b-b13a-7c93-c627cd49cc35
* Version Independent ID: 17e31a5a-7b8f-9139-c2c4-4a819e2cde7e
* Content: [New-TransportRule (ExchangePowerShell)](https://learn.microsoft.com/en-us/powershell/module/exchange/New-TransportRule?view=exchange-ps)
* Content Source: [exchange/exchange-ps/exchange/New-TransportRule.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/main/exchange/exchange-ps/exchange/New-TransportRule.md)
* Service: **exchange-powershell**
* GitHub Login: @chrisda
* Microsoft Alias: **chrisda**

**Comment by chrisda** [comment]
@riohedfield, I wouldn't make the leap that because those special tags are OK with GenerateNotification that they're also OK with  ApplyHtmlDisclaimerText, especially since the parameter description makes no such claim. I get the same error when I tried creating the disclaimer rule in the EAC, so those special tags must not be allowed.

**Comment by riohedfield** [comment]
@chrisda thanks for replying, what you just mentioned is what I already discovered, but my whole request is about to have the documentation updated with what exactly works so we don't have to guess or waste time on testing whatever comes my mind. Is it feasible?
Thank you :)

**Comment by chrisda** [comment]
@riohedfield, I can confirm that none of those special tags work. Beyond that, I know no more than what the description says:

> Disclaimer text can include HTML tags and inline cascading style sheet (CSS) tags. You can add images using the IMG tag.

**Comment by riohedfield** [comment]
@chrisda sorry but you're not answering my request. 
Let's start from this one that I've tested and works: 
%%DisplayName%%
So what else works besides this one? What else can be used as html tags/placeholders ... in ApplyHtmlDisclaimerText when creating a transport rule?
I hope my request is clear now.
It would be really helpful to have it documented.

**Comment by chrisda** [comment]
@riohedfield, I completely understand what you're asking for; I just didn't know where or how to find an answer other than suggesting that you open a support ticket and ask there.

But, thanks to your revelation about %%DisplayName%% (which wasn't even listed as a valid value for GenerateNotification!) through some chicanery and dark arts, I was able to find the following list:

- %%City%%
- %%Company%%
- %%CountryOrRegion%%
- %%Department%%
- %%DisplayName%%
- %%Fax%%
- %%FirstName%%
- %%HomePhone%%
- %%Initials%%
- %%LastName%%
- %%Manager%%
- %%MobilePhone%%
- %%Notes%%
- %%Office%%
- %%Pager%%
- %%Phone%%
- %%PostalCode%%
- %%PostOfficeBox%%
- %%StateOrProvince%%
- %%StreetAddress%%
- %%Title%%
- %%UserPrincipalName%%
- %%WindowsEmailAddress%%

I was able to create a new transport rule using all of those values.

OK?

**Comment by riohedfield** [comment]
@chrisda Awesome! and thanks a lot, can we make that list official and update the documentation with it? Or if they are in a different Microsoft article maybe you can add the reference url in a note or right under the current parameter (ApplyHtmlDisclaimerText) description?
The article will be much better and understandable if we know what that parameter supports.

**Comment by chrisda** [comment]
Yes. Documentation updated in multiple places (see links automatically added to this issue).

**Comment by riohedfield** [comment]
Thank you so much @chrisda, have a great day.
